File: dotnet/cafdemalihapa/Directories/Hosting/Feature/Automations/ProcessesMetaData/ProcessesMetaDataDirectory.cs
namespace cafdemalihapa.Directories.Hosting.Feature.Automations.ProcessesMetaData
{
    public class ProcessesMetaDataDirectory(
         AutomationsDirectory automationsDirectory
        )
    {
        public void Create()
        {
            var path = GetPath();
            Directory.CreateDirectory(path);
        }
        public string GetPath()
        {
            var directory = automationsDirectory.GetPath();
            var processesMetaDataDirectory = Path.Combine(directory, "processes-meta-data");
            return processesMetaDataDirectory;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Automations/ProcessesMetaData/ProcessesMetaDataDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Automations.ProcessesMetaData

open System.IO
open cafdemwimu.console.Directories.Hosting.Feature.Automations

type ProcessesMetaDataDirectory(automationsDirectory: AutomationsDirectory) =
    member this.Create() =
        let path = this.GetPath()
        Directory.CreateDirectory(path) |> ignore

    member _.GetPath() =
        let directory = automationsDirectory.GetPath()
        let processesMetaDataDirectory = Path.Combine(directory, "processes-meta-data")
        processesMetaDataDirectory
